apis/placement/v1alpha1/stagerollout_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",categories={fleet,fleet-placement}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StageRollout defines a group of override policies about how to override the selected cluster scope resources
+// to target clusters.
+type StageRollout struct {
```

Comment:
Hi Ryan! This will be a separate GVK, how would the users be able to use it?

Reply:
I assume that since there's a separate `RolloutStrategyType`, we would also need a config field like `RollingUpdateConfig` in the CRP API, with the field itself being a selector that selects a GVK instance; is my understanding correct?